utilities/include/vec3.hpp

```diff
@@ -1,212 +1,59 @@
 #ifndef VEC3_HPP
 #define VEC3_HPP
 
-
 #include <iostream>
-#include <cmath> //constexpr is compiler specific
-class vec3{
-    float x,y,z;
-    public:
-        constexpr vec3() : x {0}, y {0}, z {0} {}
-
-        //Cartesian coordinate Constructor
-        constexpr vec3(float X,float Y,float Z) : x {X}, y {Y}, z {Z} {}
-
-        //norm function
-        #ifndef __GNUC__
-
-        float norm() const {
-            return sqrt(x * x + y * y + z * z);
-        }
-
-        #endif
-
-        #ifdef __GNUC__
-        
-        constexpr float norm() const {
-            return sqrt(x * x + y * y + z * z);
-        }
-
-        #endif
-
-
-
-        //unit vector function
-        constexpr vec3 unit() const{
-            return *this/this->norm();
-        }
-
-        //getters
-        constexpr float X() const{
-            return x;
-        }
+#include <cmath>
+#include <type_traits>
```

Comment:
It would be good if you mentioned what we need this header for. Something like `// for std::is_constant_evaluated` would be nice.